src/bin/pg_autoctl/supervisor.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * src/bin/pg_autoctl/supervisor.h
+ *   Utilities to start/stop the pg_autoctl services.
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+#ifndef SUPERVISOR_H
+#define SUPERVISOR_H
+
+#include <inttypes.h>
+#include <signal.h>
+
+/*
+ * start and stop function used in the struct Service.
+ *
+ * start functions get passed a context that is manually provided in the
+ * structure, typically a Monitor struct pointer for the monitor service and a
+ * Keeper struct pointer for a keeper service.
+ *
+ * stop functions get passed a context that is a pointer to the Service struct
+ * definition of the service being asked to stop, and we use a void * data type
+ * here to break out of a mutual recursive definition.
+ *
+ * reload functions get passed a context that is a pointer to the Service
+ * struct definition of the service being asked to stop.
+ */
+typedef bool (*ServiceStartFunction)(void *context, pid_t *pid);
+typedef bool (*ServiceStopFunction)(void *context);
+typedef void (*ServiceReloadFunction)(void *context);
```

Comment:
As far as I can tell the typedefs are not needed anymore now for the bottom two for consistency sake I think it's nicer to also move the startFunction inline into the struct.
```suggestion
```

Reply:
Done now. Thanks again!